db/database.go

```diff
@@ -492,6 +520,9 @@ func (context *DatabaseContext) Close() {
 	context.mutationListener.Stop()
 	context.changeCache.Stop()
 	context.ImportListener.Stop()
+	if context.Heartbeater != nil {
+		context.Heartbeater.Stop()
+	}
```

Comment:
Already doing nil checks inside stop, probably safer/easier to use if it's down at that level.

```suggestion
	context.Heartbeater.Stop()
```